src/_tutorials/language/futures.md

```diff
@@ -70,15 +70,17 @@ Such functions return their value using a [Future.][Future]
 
 ## What is a Future? {#what-is-a-future}
 
-A Future represents a means for getting a value sometime in the future. When a
-function that returns a Future is invoked, two things happen:
+A Future represents an asynchronous operation that, optionally,
+returns a value. When a function that returns a Future is invoked,
```

Comment:
_Expression_ sounds too lightweight to me — like it's a small Dart unit of code and you can't even have a Future execute a whole statement.

Reply:
While its technically true that a Future always returns a value, a void value is one that you aren't supposed to use. The fact that there's something there is more of an implementation detail, which is fully specified so that you can grab it in an emergency, and so that we don't have to jump through some technical hoops completely guaranteeing its vacuousness.

Also agree that expression is too lightweight for this.

You could say "optionally returns a usable value," or "optionally passing along a value" (since a `Future<void>` indicates that the value is not supposed to be read, its not really being "passed along"?) or "optionally including a usable result" ?

Reply:
Maybe "returns a non-void value"?

Reply:
That works for me!

Reply:
Actually, I decided to use "returns a value (which can be `void`)." That simplifies some language that appears a few lines later: "Later... completes with a value or with an error."